src/krylov/pcg.c

```diff
@@ -703,12 +789,28 @@ hypre_PCGSolve( void *pcg_vdata,
          (pcg_data -> converged) = 1;
          break;
       }
-
+      /* gamma should generally be greater than 0 for spd prec and nonzero r */
       if (! (gamma > HYPRE_REAL_MIN) )
       {
          hypre_error_w_msg(HYPRE_ERROR_CONV, "Subnormal gamma value in PCG");
-
-         break;
+         if (!skip_break)
+         {
+            break;
+         }
+         else if (! (gamma >= HYPRE_TRUE_MIN))
```

Comment:
Same comment here as above on wanting to always do these tests to get more error message info.